src/rebuild/initiator.c

```diff
@@ -53,6 +62,7 @@ struct puller_iter_arg {
 	rebuild_obj_iter_cb_t		obj_cb;
 	daos_handle_t			cont_hdl;
 	struct rebuild_root		*cont_root;
+	struct rebuild_snapshots	*snaps;
```

Comment:
Isn't the puller_iter_arg per container? I don't quite see why a separate allocation and refcount are required here.

Reply:
No, puller_iter_arg is not per container, see rebuild_obj_handler(), it is just an iteration argument. Actually I am not sure what is your question here?  Adding a refcount here is for snapshots can be freed by itself. Note: we do not open/close container during rebuild.

Reply:
See my comments in puller_cont_iter_cb().